themes/xresources/theme.lua.in

```diff
@@ -0,0 +1,222 @@
+---------------------------------------------
+-- Awesome theme which follows xrdb config --
+---------------------------------------------
+
+local awful = require("awful")
+
+local xresources = {}
+
+xresources.fallback = {
+  --black
+  ["0"] = '#000000',
+  ["8"] = '#465457',
+  --red
+  ["1"] = '#cb1578',
+  ["9"] = '#dc5e86',
+  --green
+  ["2"] = '#8ecb15',
+  ["10"] = '#9edc60',
+  --yellow
+  ["3"] = '#cb9a15',
+  ["11"] = '#dcb65e',
+  --blue
+  ["4"] = '#6f15cb',
+  ["12"] = '#7e5edc',
+  --purple
+  ["5"] = '#cb15c9',
+  ["13"] = '#b75edc',
+  --cyan
+  ["6"] = '#15b4cb',
+  ["14"] = '#5edcb4',
+  --white
+  ["7"] = '#888a85',
+  ["15"] = '#ffffff',
+  --
+  bg  = '#0e0021',
+  fg  = '#bcbcbc',
+}
+
+--- Get current base colorscheme from xrdb.
+-- Returns a table with keys 'bg', 'fg' and '0'..'15'
+function xresources.get_current_theme()
+  local colors = {}
+  local output = io.popen("xrdb -query")
+  local query = output:read('*a')
+  output:close()
+  for i,color in string.gmatch(query, "%*color(%d+):[^#]*(#[%a%d]+)") do
+    colors[i] = color
+  end
+  if not colors["15"] then
+      colors = awful.util.table.clone(xresources.fallback)
+  end
+  colors.bg = string.match(query, "*background:[^#]*(#[%a%d]+)") or xresources.fallback.bg
+  colors.fg = string.match(query, "*foreground:[^#]*(#[%a%d]+)") or xresources.fallback.fg
+  return colors
+end
+
+
+--- Compute resulting fontsize applying dpi value from xrdb.
+-- @tparam size int Font size
+function xresources.compute_fontsize(size)
+  if not xresources.dpi then
+    local output = io.popen("xrdb -query")
+    local query = output:read('*a')
+    xresources.dpi = tonumber(string.match(query, "dpi:[%s]+([%d]+)"))
```

Comment:
Ah. Maybe `xdpyinfo` could be used then instead?

This setting can (presumably) also be different per screen, could this be handled?